src/datasets/datasets.controller.ts

```diff
@@ -536,15 +556,29 @@ export class DatasetsController {
     @Body()
     createDatasetObsoleteDto:
       | CreateRawDatasetObsoleteDto
-      | CreateDerivedDatasetObsoleteDto,
+      | CreateDerivedDatasetObsoleteDto
+      | CreateDatasetDto,
   ): Promise<OutputDatasetObsoleteDto> {
     // validate dataset
+    let dtoType;
+    switch (createDatasetObsoleteDto.type) {
```

Comment:
We can maybe extract this in it's own function and reuse it in all the places needed.

Reply:
Had to revert this change, the return Dto type (e.g. Create, Update and PartialUpdate) is different in each of the three instances this happens, so the logic is not identical

Reply:
Yes I see it now. OK then maybe we can keep it as it is for now as this will be refactored and cleaned up little bit soon where we will use mostly the standardized DTOs for all the dataset endpoints accordingly